lib/kafka/connection.rb

```diff
@@ -88,10 +93,10 @@ def send_request(request)
 
         @correlation_id += 1
 
-        write_request(request, notification)
+        @operation.write_request(request, notification, @correlation_id)
 
         response_class = request.response_class
-        wait_for_response(response_class, notification) unless response_class.nil?
+        @operation.wait_for_response(response_class, notification, @correlation_id) unless response_class.nil?
```

Comment:
Would it be possible to do the handshaking using only calls to `write_request` and `wait_for_response` or do you _need_ lower level writes and reads? I think it would be better to handle this at the client level if possible.